cli/src/base.ts

```diff
@@ -100,10 +100,18 @@ export abstract class BaseCommand extends Command {
     };
   }
 
+  loadEnvFile(path: string) {
+    const env = dotenv.config({ path });
+    if (env.parsed?.['XATA_API_KEY']) {
+      this.apiKeyLocation = 'dotenv';
+      this.apiKeyDotenvLocation = path;
+    }
+  }
+
   async init() {
     if (process.env.XATA_API_KEY) this.apiKeyLocation = 'shell';
-    const env = dotenv.config({ path: this.dotenvLocation });
-    if (env.parsed?.['XATA_API_KEY']) this.apiKeyLocation = 'dotenv';
+    this.loadEnvFile('.env.local');
+    this.loadEnvFile('.env');
```

Comment:
I don't think this is enough.

As I mentioned in the thread, it's best that `.env.*` takes priority instead of just local.

If you check the Next.js docs, it's what they do too. They even recommend different patterns such as `.env.development` and `.env.production`

If the user has multiple, they need to configure, that's ok. But if they have only `.env.development` will not work with this implementation, and it's a decent use-case.

Reply:
@atilafassina from a CLI perspective there's no "environment". You are working locally.

Reply:
For the future: I don't think we want to implement this. Next.js uses `NODE_ENV` to load those files, and `NODE_ENV` typically is used inside a Node.js process. In our case you are invoking the CLI from a shell, not from a Node.js process. Besides, you are either running the CLI locally or from CI. If you are using the CLI locally, then `local` is assumed. And if you are running the CLI from CI then you can just set environmental variables in the shell before running the CLI.